Unit test course component

lucy811 opened the pull request:
1.Fixed unit test errors in other components.
2.Wrote unit tests in courses.component.spec.ts and courses-add.component.spec.ts
<img width="656" alt="unit_test" src="https://user-images.githubusercontent.com/25615945/33899529-6a40ebc4-df31-11e7-8974-95169613f83c.png">

3. npm run v-lint-fix
<img width="650" alt="v-lint" src="https://user-images.githubusercontent.com/25615945/33899537-6ff0eed4-df31-11e7-94bb-7caa0bc8f310.png">

lmmrssa commented:
@lucy811 could you please rebase the branch with master. We had fixed the travis break. So your test might build after that.

lmmrssa commented on src/app/nation/nation.component.spec.ts:
```diff
@@ -1,23 +1,26 @@
 import { ReactiveFormsModule, FormsModule } from '@angular/forms';
+import { MaterialModule } from '../shared/material.module';
 import { FormErrorMessagesComponent } from '../shared/form-error-messages.component';
+import { Jsonp, JsonpModule } from '@angular/http';
```
we are trying to replace these deprecated elements from newer ones. So could you explore for @angular/common/http equivalent elements

lucy811 commented:
1. I have fixed new unit test errors in other components after rebased the branch with master.
2. I have deleted Jsonp and JsonpModule in nation.component.spec.ts

nirojdyola commented on src/app/courses/add-courses/courses-add.component.spec.ts:
```diff
@@ -1,33 +1,97 @@
+    });
+  });
+
+  // test addCourse()
```
unnecessary comments can be removed.Could you please check it once ?

nirojdyola commented on src/app/resources/resources.component.spec.ts:
```diff
@@ -49,16 +50,19 @@ describe('ResourcesComponent', () => {
   });
 
   // test getRating()
```
unncessary comments can be removed. Could you please check it once ?

nirojdyola commented on src/app/users/users.component.spec.ts:
```diff
@@ -94,10 +94,9 @@ describe('Users', () => {
         initSpy = spyOn(comp, 'initializeData').and.callFake(() => { } ),
         couchSpy = spyOn(couchService, 'put').and.returnValue(Promise.resolve({}));
-      comp.deleteRole({ name: 'Test', roles: [ 'one', 'two', 'three' ] }, 1, testEvent);
+      comp.deleteRole({ name: 'Test', roles: [ 'one', 'two', 'three' ] }, 1 );
```
extra space after 1. Could you please remove it ?

lucy811 commented:
@nirojdyola I have deleted unnecessary comments and extra space.

lucy811 commented:
Added imports:[RouterTestingModule.withRoutes([{ path: 'courses', component: CoursesAddComponent}])] in courses-add.component.spec.ts in order to spyon router in test addCourse().

paulbert commented:
Hi @lucy811 it worked to simply update the `debugOperator` in `app.component.spec.ts` so that is fixed.  There is one other thing for you to do to update the tests to the changes on the `master`: every time a spy on `CouchService` returned a `Promise.resolve` you will have to:
1. `import { of } from 'rxjs/observable/of';` at the top of the spec file (once per file)
2. Replace `Promise.resolve` with `of` using the same object as the parameter of the function

Right now the tests for me are not completing, but once this is changed it should work.  Let us know if you run into any other issues!

lucy811 commented:
1. Imported { of } from 'rxjs/observable/of' in each .spec file.
2. Replaced Promise.resolve with of
3. Replaced Promise.reject with Rx.Observable.throw
4. Used 'ng lint --fix' to check errors
